skimage/util/dtype.py

```diff
@@ -32,7 +32,73 @@
 _supported_types = list(dtype_range.keys())
 
 
-def dtype_limits(image, clip_negative=False):
+def _check_precision_loss(dtypeobj_in, dtypeobj_out):
+    """Check if conversion between images will incur loss of precision.
+
+    Generally speaking, the output object needs to have more significant bits
+    than the input kind.
+
+    Details:
+        float -> signed/unsigned ints will always return true.
+        signed/unsigned -> float of same number of bytes will return true.
+
+    Parameters
+    ----------
+    dtypeobj_in: np.dtype
+        dtype of the input image.
+    dtypeobj_out: np.dtype
+        dtype of the output image.
+    output_warning: bool, optional
+        Outputs a warning using `warn` if operation will incur loss of
+        precision.
+    int_same_size_ok: bool, optional
+        Should conversion between integers of the same same (and signedness)
+        be considered lossy.
+
+    """
+    dtypeobj_in = np.dtype(dtypeobj_in)
+    dtypeobj_out = np.dtype(dtypeobj_out)
+    kind_in = dtypeobj_in.kind
+    kind_out = dtypeobj_out.kind
+    itemsize_in = dtypeobj_in.itemsize
+    itemsize_out = dtypeobj_out.itemsize
+
+    has_loss = False
+
+    if kind_in != 'b' and kind_out == 'b':
```

Comment:
There seems to be some more generic pattern here.

Can we check only the itemsize_in vs itemsize_out?

Reply:
plus minor logic for signed ranges

Reply:
seems like i can just treat it as 'u' and do something like `kind in ['b', 'u']`

Reply:
Seems like I cannot.
```
In [1]: import numpy as np

In [2]: np.dtype(np.uint8).itemsize
Out[2]: 1
```

Reply:
So `bool` is the exception that needs to be checked for, because it is atypically sized.